huaweicloud/services/cfw/resource_huaweicloud_cfw_ip_address_group_member.go

```diff
@@ -0,0 +1,249 @@
+		"address_items": []map[string]interface{}{item},
+	}
+	return bodyParams
+}
+
+func resourceIPAddressGroupMemberRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getIPAddressGroupMember: Query the CFW IP address group member detail
+	var (
+		getIPAddressGroupMemberHttpUrl = "v1/{project_id}/address-items"
+		getIPAddressGroupMemberProduct = "cfw"
+	)
+	getIPAddressGroupMemberClient, err := cfg.NewServiceClient(getIPAddressGroupMemberProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	getIPAddressGroupMemberPath := getIPAddressGroupMemberClient.Endpoint + getIPAddressGroupMemberHttpUrl
+	getIPAddressGroupMemberPath = strings.ReplaceAll(getIPAddressGroupMemberPath, "{project_id}",
+		getIPAddressGroupMemberClient.ProjectID)
+
+	getIPAddressGroupMemberqueryParams := buildGetIPAddressGroupMemberQueryParams(d)
+	getIPAddressGroupMemberPath += getIPAddressGroupMemberqueryParams
+
+	getIPAddressGroupMemberOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getIPAddressGroupMemberResp, err := getIPAddressGroupMemberClient.Request("GET", getIPAddressGroupMemberPath,
+		&getIPAddressGroupMemberOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving IPAddressGroupMember")
+	}
+
+	getIPAddressGroupMemberRespBody, err := utils.FlattenResponse(getIPAddressGroupMemberResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(utils.PathSearch("data.records[0].item_id", getIPAddressGroupMemberRespBody, nil).(string))
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("group_id", utils.PathSearch("data.set_id", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("name", utils.PathSearch("data.records[0].name", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("address", utils.PathSearch("data.records[0].address", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("address_type", utils.PathSearch("data.records[0].address_type", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("description", utils.PathSearch("data.records[0].description", getIPAddressGroupMemberRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetIPAddressGroupMemberQueryParams(d *schema.ResourceData) string {
+	res := "?offset=0&limit=10"
+
+	res = fmt.Sprintf("%s&set_id=%v", res, d.Get("group_id"))
+
+	res = fmt.Sprintf("%s&address=%v", res, d.Get("address"))
+
+	if res != "" {
+		res = "?" + res[1:]
+	}
+	return res
+}
+
+func resourceIPAddressGroupMemberDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteIPAddressGroup: Delete an existing CFW IP address group
+	var (
+		deleteIPAddressGroupHttpUrl = "v1/{project_id}/address-items/{id}"
+		deleteIPAddressGroupProduct = "cfw"
```

Comment:
it is better to change the name to deleteIPAddressGroupMemberHttpUrl, so that it is  consistent with other methods

Reply:
fixed